### PR TITLE
[sei] Add checks to avoid overriding event.original if present

### PR DESCRIPTION
**Pull request by marc-gr**
## Proposed commit message

Add checks to avoid overriding event.original if present.
On some setups (ie Logstash sends data to `ES`) `event.original` might be already present in the event body. Currently most of our integrations do not have a null check for it before trying to set it, leading to errors on these circumstances. By adding the required checks we improve handling in this setups.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes https://github.com/elastic/integrations/issues/8146

### CODE REVIEW

File: packages/cef/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -26,16 +26,16 @@ processors:
...
       target_field: source.as
   - geoip:
       field: destination.ip
       database_file: GeoLite2-ASN.mmdb
       ignore_missing: true
       properties:
-      - asn
-      - organization_name
+        - asn
+        - organization_name
```
Comment (kcreddy) [inline packages/cef/data_stream/log/elasticsearch/ingest_pipeline/default.yml:38]: These changes doesn't seem to contain `event.original`. However, it is present in the input config.

**Review by kcreddy — COMMENTED** [review]
`cyberark_pta`, `trendmicro` also has it in the input config like `imperva`

**Comment by efd6** [comment]
When you commit this, please use this as the git-generate script.
```
#!/bin/bash

for f in packages/*; do
    pkg=$(yq 'select(.owner.github | contains("elastic/security-external-integrations")) .name' < ${f}/manifest.yml 2>/dev/null)
    if [[ $pkg == "" ]]; then
        continue
    fi

    (
        cd packages/${pkg}
        glob="./data_stream/**/elasticsearch/ingest_pipeline/default.yml"
        files=$(echo $glob | tr " " "\n")
        for file in $files; do
            if grep -q "field: event.original" $file; then
                yq e -i '(.processors | .[] | select(.rename.target_field == "event.original")) *= {"rename":{"field": "message", "target_field": "event.original", "ignore_missing": true, "if": "ctx.event?.original == null"}}' $file
                yq e -i '(.processors | .[] | select(.set.field == "event.original" and .set.copy_from != "_temp_.result._raw")) = {"rename":{"field": "message", "target_field": "event.original", "ignore_missing": true, "if": "ctx.event?.original == null"}}' $file
            fi
        done
        if ! git diff --quiet --exit-code .; then
            elastic-package changelog add --next minor --type enhancement --link elastic#8269 --description "Improve 'event.original' check to avoid errors if set."
            elastic-package check
        fi
    )
done
```
It encapsulates the full resultant commit into one operation and is host agnostic. Note that this changes more files than are changed here, so it may be worth checking that all the SEI packages have been touched, or whether there are packages that the integrated script touches that it shouldn't.

**Comment by marc-gr** [comment]
> Ooop. It appears that not all of the changes here are mechanical. Is that correct?

Correct. The automated way covered way more than required, I had to manually validate and do some changes. That is why I ended up removing the `git-generate` scripts since they were unable to replicate the commits 100%.

**Comment by efd6** [comment]
Thanks. Make sure the git-generate text is not in the final merge commit then.

**Review by kcreddy — COMMENTED** [review]
Are we targeting only the ingest pipeline changes in this PR?
Following packages/datastreams need to be updated in input config.
```
carbonblack_edr 
cef 
cyberark_pta
imperva 
panw 
system.security 
thycotic_ss 
trendmicro 
windows.applocker_exe_and_dll 
windows.applocker_msi_and_script 
windows.applocker_packaged_app_deployment 
windows.applocker_packaged_app_execution 
windows.forwarded 
windows.powershell 
windows.powershell_operational 
windows.sysmon_operational 
winlog 
```

**Comment by marc-gr** [comment]
> Are we targeting only the ingest pipeline changes in this PR? Following packages/datastreams need to be updated in input config.
> 
> ```
> carbonblack_edr 
> cef 
> cyberark_pta
> imperva 
> panw 
> system.security 
> thycotic_ss 
> trendmicro 
> windows.applocker_exe_and_dll 
> windows.applocker_msi_and_script 
> windows.applocker_packaged_app_deployment 
> windows.applocker_packaged_app_execution 
> windows.forwarded 
> windows.powershell 
> windows.powershell_operational 
> windows.sysmon_operational 
> winlog 
> ```

The changes are only required for packages that populate `event.original` on the pipeline. This is to avoid issues when events come to the pipeline from Logstash instead of Elastic Agent, because it pre populates `event.original`, so we need to add the safety check to avoid errors. If the population happens on the input side, the pipeline already assumes `event.original` to be there, so in this case it will behave normally with these kind of set ups.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**